inc/saiswitch.h

```diff
@@ -2776,6 +2776,20 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_ECMP_MEMBER_COUNT,
 
+    /**
+     * @brief Enable/disable Clause 22 MDIO access
+     *
+     * set mdio access type (clause 45 or clause 22) before calling sai_switch_mdio_read_fn and sai_switch_mdio_write_fn
+     *
+     * FALSE - sai_switch_mdio_read_fn and sai_switch_mdio_write_fn use clasue 45 mdio access.
+     * TRUE - sai_switch_mdio_read_fn and sai_switch_mdio_write_fn use clasue 22 mdio access.
```

Comment:
could there be other clauses more in the future ? maybe this should be enum?, i think enum would be better here, since bool value should be used for something that is enabled or not, and here you have explicit types 22 and 45 to set

Reply:
MDIO was originally defined in Clause 22 of IEEE RFC802.3. Clause 45 of the 802.3ae specification provided the ability to access 65,536 registers in 32 different devices on 32 different ports. 802.3ae is 20 years old. There is no new MDIO access mode in the last 20 years. MDIO frame format starts with 32 bits of logical 1 preamble and 2 bits of start of frame. The start of frame is 01 for clause 22 and 00 for clause 45. If the first bit of the start frame has to be logical 0, then no new MDIO frame format will be possible.

Reply:
i see, but it still would make sense to make it as enum, like for example you would have ipv4 and ipv6, it's better specify it as enum rather then bool, maybe lets ask community on this

Reply:
Changed to attribute SAI_SWITCH_ATTR_MDIO_ACCESS_TYPE and enum sai_switch_mdio_clause_t